src/webnn_native/dml/GraphDML.cpp

```diff
@@ -975,12 +978,23 @@ namespace webnn_native { namespace dml {
             std::vector<float> constant(length, 2);
             auto pow = ::dml::Pow(input, BindingConstant(DML_TENSOR_DATA_TYPE_FLOAT32, inputDims,
                                                          constant.data(), sizeof(float) * length));
-            auto avgPool2d =
-                ::dml::AveragePooling(pow, strides, windowSizes, startPadding, endPadding, false);
+            auto avgPool2d = ::dml::AveragePooling(pow, strides, windowSizes, startPadding,
+                                                   endPadding, false, outputShape);
             output = ::dml::Sqrt(avgPool2d);
         } else if (pool2d->GetType() == op::Pool2dType::kMaxPool2d) {
+            // Update the output shape for dilations.
+            if (dilations[0] != 1 || dilations[1] != 1) {
```

Comment:
Only Max pool on DML backend support dilation, maybe the dilation will be removed from Spec, do we support it currently?

Reply:
The case for maxpool2d with dilations on debug version will crash due to the mismatch when checking outshape, so I disable it.

Reply:
Should we remove the workaround?

Reply:
Does it disable the dilation maxpool as we discussion?

Reply:
> No, now we use outputShape to support roundingType, removing these will cause regression because dilated maxpool will get wrong outputshape.

In the past, our dilated maxpool can pass, if I set utputShape to support roundingType in this PR, I must ensure there is no regression, so I have to support the dilated maxpool here. @fujunwei

Reply:
Regrssion is ok for dilated maxpool, i think the workarround can be add later.

Reply:
But regression will cause CI failure. @fujunwei